.github/workflows/CI.yml

```diff
@@ -78,11 +97,78 @@ jobs:
       shell: bash
       run: poetry run python -m pip install --upgrade pip
 
-    # Install dependencies
+    # Install dependencies - no soft dependencies
     - name: Install dependencies
       run: |
         poetry install
 
+    # Run tests
+    - name: Test with pytest
+      run: |
+        poetry run python -m pytest tests -vv
+
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-latest]
```

Comment:
Windows is missing or? And other python dependenc

Reply:
simply prior status quo.

I'll start adding windows and broader versions when it is easier to manage soft deps